pandas/src/generate_code.py

```diff
@@ -2286,6 +2292,45 @@ def generate_put_template(template, use_ints=True, use_floats=True,
         output.write(func)
     return output.getvalue()
 
+def generate_put_selection_template(template, use_ints=True, use_floats=True,
+                                    use_objects=False, use_datelikes=False):
+    floats_list = [
+        ('float64', 'float64_t', 'float64_t', 'np.float64', 'nan', 'np.inf'),
+        ('float32', 'float32_t', 'float32_t', 'np.float32', 'nan', 'np.inf'),
+    ]
+    ints_list = [
+        ('int8',  'int8_t',  'int8_t', 'np.int8', 'iNaT', _int8_max),
```

Comment:
The current implementation just casts them to float64 before running them though the cython functions. However if you look at the other issue that this fixes (#6620) there are some edge-cases with really large integers where the int->float->int conversion causes a loss of precision. Since min and max are just selecting values, we can fix those edge cases if we keep the types the same for these ops.